### PR TITLE
[MLIR][Python] enhance python ir printing with pringing flags

### PR DESCRIPTION
**Pull request by qingyunqu**
Close https://github.com/llvm/llvm-project/pull/65854

### CODE REVIEW

**Review by ftynse — APPROVED** [review]

**Comment by ftynse** [comment]
A test would be nice.

**Comment by qingyunqu** [comment]
> A test would be nice.

Done.

File: mlir/test/python/pass_manager.py [hunk 1]
```diff
@@ -342,6 +342,33 @@ def testPrintIrBeforeAndAfterAll():
...
+            }
+          }
+        """
+        )
+        pm = PassManager.parse("builtin.module(canonicalize)")
+        ctx.enable_multithreading(False)
+        pm.enable_ir_printing(large_elements_limit=2)
+        # CHECK: // -----// IR Dump After Canonicalizer (canonicalize) //----- //
+        # CHECK: module {
+        # CHECK:   func.func @main() -> tensor<3xi64> {
```
Comment (ftynse) [inline mlir/test/python/pass_manager.py:364]:
```suggestion
```

File: mlir/test/python/pass_manager.py [hunk 2]
```diff
@@ -342,6 +342,33 @@ def testPrintIrBeforeAndAfterAll():
...
+        # CHECK:     %[[CST:.*]] = arith.constant dense_resource<__elided__> : tensor<3xi64>
+        # CHECK:     return %[[CST]] : tensor<3xi64>
+        # CHECK:   }
+        # CHECK: }
```
Comment (ftynse) [inline mlir/test/python/pass_manager.py:368]:
```suggestion
```

It is sufficient to check for the specific behavior the flag is changing: large resource elision. The rest is likely to get spuriously broken when syntax changes.

**Review by ftynse — APPROVED** [review]
Let me know if you need help landing this.

**Comment by qingyunqu** [comment]
> Let me know if you need help landing this.

Sure, it's better if you could help merge this PR.

**Pull request merged**